webapp/src/Service/DOMJudgeService.php

```diff
@@ -1675,4 +1681,79 @@ public function getAllowedLanguagesForContest(?Contest $contest) : array {
             ->getQuery()
             ->getResult();
     }
+
+    /**
+     * Returns either the next strictly higher version or false when nothing is found/requested.
+     */
+    public function checkNewVersion(): string|false {
+        if ($this->config->get('check_new_version', false) === UpdateStrategy::Strategy_none) {
+            return false;
+        }
+        // The local version is something like "x.y.z / commit hash", e.g. "8.4.0DEV/4e25adb13" for development
+        // or 8.3.2 for a released version
+        // In case of development we remove the commithash for some anonymity but keep the DEV to not count those as the (possibly) released version
+        $localVersionString = strtok($this->domjudgeVersion, "/");
+        $localVersion = explode(".", $localVersionString);
+        if (count($localVersion) !== 3) {
+            // Unknown version, someone might have locally modified and used their own versioning
+            return false;
+        }
+        $versionUrl = 'https://versions.domjudge.org';
+        $options = ['http' => ['method' => 'GET', 'header' => "User-Agent: DOMjudge#" . $this->domjudgeInstallMethod . "/" . $localVersionString . "\r\n"]];
+        $context = stream_context_create($options);
+        $response = @file_get_contents($versionUrl, false, $context);
+        if ($response === false) {
+            return false;
+        }
+        // Assume we get a one-level unordered JSON list with the released versions e.g. ["10.0.0", "9.11.0", "12.0.12", "10.0.1"]
+        $versions = json_decode($response, true);
+        natsort($versions);
+        $versions = array_reverse($versions);
+        preg_match("/\d.\d.\d/", $this->domjudgeVersion, $matches);
+        $extractedLocalVersionString = $matches[0];
+        if ($this->config->get('check_new_version', false) === UpdateStrategy::Strategy_incremental) {
+            /* Steer towards the nearest new patch release first
```

Comment:
I might be alone with this opinion, but I think we should tell users the most recent version instead.

Feel free to close

Reply:
I think that having both strategies is nice, if we actually do the bugfix backports. If we do not plan to more actively do these, then I agree that we should guide users towards the latest version.

Reply:
I think we should do backports/releases more indeed and I think offering both patch and major/minor release updates makes sense. Why can't we display both at the same time?

Reply:
Have we people asking for backport releases? They are a lot of work and I'm not sure whether it's worth the effort